### PR TITLE
feat: chainbridge linked to new binary for CI test

### PR DESCRIPTION
**Pull request by wangminqi**
This PR is for linking chainbridge ts test to latest binary.

In order to properly view this PR. Request code review in 
Chainbridge Binary Change
https://github.com/litentry/ChainBridge/pull/2
Chainbridge Solidity Change
https://github.com/litentry/chainbridge-solidity/pull/1

### CODE REVIEW

File: docker/bridge.dockerfile [hunk 1]
```diff
@@ -1,3 +1,3 @@
 FROM golang:1.18
 
-RUN go install github.com/litentry/ChainBridge/cmd/chainbridge@dev
+RUN go install github.com/litentry/ChainBridge/cmd/chainbridge@minqi-dev
```
Comment (BillyWooo) [inline docker/bridge.dockerfile:3]: Why don't use `dev` branch?

Comment (wangminqi) [reply docker/bridge.dockerfile:3]: dev is phala's smart contract. minqi-dev is our new smart contract.

**Comment by wangminqi** [comment]
In order to properly view this PR. Request code review in 
Chainbridge Binary Change
https://github.com/litentry/ChainBridge/pull/2
Chainbridge Solidity Change
https://github.com/litentry/chainbridge-solidity/pull/1

File: ts-tests/tests/setup-bridge.ts [hunk 2]
```diff
@@ -123,9 +123,9 @@ async function setupCrossChainTransfer(
     await eConfig.erc20.mint(eConfig.wallets.eve.address, toWei('100000'));
     await eConfig.erc20.mint(eConfig.erc20Handler.address, toWei('300'));
     await eConfig.bridge.adminSetResource(eConfig.erc20Handler.address, destResourceId, eConfig.erc20.address);
-    await eConfig.bridge.adminSetDecimals(eConfig.erc20Handler.address, eConfig.erc20.address, 18, 12, opts);
+    // await eConfig.bridge.adminSetDecimals(eConfig.erc20Handler.address, eConfig.erc20.address, 18, 12, opts);
```
Comment (arieljbirnbaum) [inline ts-tests/tests/setup-bridge.ts:126]: If this ain't needed, please remove it altogether :)

File: ts-tests/tests/bridge.test.ts [hunk 3]
```diff
@@ -82,16 +82,24 @@ describeCrossChainTransfer('Test Cross-chain Transfer', ``, (context) => {
         const provider = context.ethConfig.wallets.alice.provider;
         const currentBlock = await provider.getBlockNumber();
         await sleep(15);
+
+        let voteTransactionCount = 0;
```
Comment (arieljbirnbaum) [inline ts-tests/tests/bridge.test.ts:86]: What's this counter for? The loop stops after the first `voteTransaction` anyway, so at the `if (voteTransactionCount === 1) {` part `voteTransactionCount` will _always_ equal 1 :thinking:

File: ts-tests/tests/bridge.test.ts [hunk 4]
```diff
@@ -82,16 +82,24 @@ describeCrossChainTransfer('Test Cross-chain Transfer', ``, (context) => {
...
+                    if (decodedInput.name === 'voteProposal') {
+                        voteTransactionCount++;
+                        // We have threshold = 1 in this. So it really does not matter
+                        if (voteTransactionCount === 1) {
+                            const receipt = await provider.getTransactionReceipt(tx.hash);
+                            assert.equal(1, receipt.status, 'Vote transaction should succeeds, yet the execution fails');
+                            // So the execution fails means balance no change
+                            assert.equal(handlerBalance.toString(), (await context.ethConfig.erc20.balanceOf(context.ethConfig.erc20Handler.address)).toString())
+                            return;
+                        }
```
Comment (arieljbirnbaum) [inline ts-tests/tests/bridge.test.ts:102]: I don't know enough about the bridge to comment on this :sweat: 
@BillyWooo ?

Comment (wangminqi) [reply ts-tests/tests/bridge.test.ts:102]: Originally, Ethereum smart contract will trigger two steps after receiving a proposal from Substrate to Relayer.
1.All relayers will make a proposal/vote for proposal until vote threshold passed.
2. One of relayers try make a seperate transaction to executed transaction.

The new smart contract (which this PR for) will:
1. relayer make/vote for a proposal, in the same transaction checking if passed the threshold and execute accordingly.

The change above is for the adjustment of new logic for test, where executeProposal is no longer existing.

Comment (wangminqi) [reply ts-tests/tests/bridge.test.ts:102]: Yes, it is a rather sloppy change. Simply for the consistency of test behavior.

Comment (wangminqi) [reply ts-tests/tests/bridge.test.ts:102]: executeProposal is no longer existing. It is "replaced" by the last voteProposal transaction of all relayers.

File: ts-tests/tests/bridge.test.ts [hunk 5]
```diff
@@ -82,16 +82,24 @@ describeCrossChainTransfer('Test Cross-chain Transfer', ``, (context) => {
...
                     const tx = block.transactions[j];
                     const decodedInput = inter.parseTransaction({ data: tx.data, value: tx.value });
-                    if (decodedInput.name === 'executeProposal') {
-                        const receipt = await provider.getTransactionReceipt(tx.hash);
-                        assert.equal(0, receipt.status, 'Expect the transaction fail, it actually succeeds');
-                        return;
+                    if (decodedInput.name === 'voteProposal') {
+                        voteTransactionCount++;
+                        // We have threshold = 1 in this. So it really does not matter
+                        if (voteTransactionCount === 1) {
```
Comment (arieljbirnbaum) [inline ts-tests/tests/bridge.test.ts:96]: :nail_care: the intent can be made a bit more explicit by

```suggestion
                        if (voteTransactionCount === approvalThreshold) {
```

with a `const approvalThreshold = 1` earlier

File: ts-tests/tests/bridge.test.ts (hunk 4 above)
Comment (arieljbirnbaum) [reply ts-tests/tests/bridge.test.ts:102]: Makes sense now; thanks for the explanation!

**Review by arieljbirnbaum — APPROVED** [review]
looks ok to me

**Comment by wangminqi** [comment]
This PR is not urgent and we will not change token bridge code in short time.

**Pull request closed without merging**